client/my-sites/domains/domain-management/dns/dns-records-list.jsx

```diff
@@ -0,0 +1,232 @@
+import { connect } from 'react-redux';
+import MaterialIcon from 'calypso/components/material-icon';
+import { domainConnect } from 'calypso/lib/domains/constants';
+import DnsRecordsListHeader from 'calypso/my-sites/domains/domain-management/dns/dns-records-list-header';
+import { addDns, deleteDns } from 'calypso/state/domains/dns/actions';
+import { isDeletingLastMXRecord } from 'calypso/state/domains/dns/utils';
+import { errorNotice, removeNotice, successNotice } from 'calypso/state/notices/actions';
+import DnsRecordData from './dns-record-data';
+
+class DnsRecordsList extends Component {
+	static propTypes = {
+		dns: PropTypes.object.isRequired,
+		selectedDomainName: PropTypes.string.isRequired,
+		selectedSite: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ).isRequired,
+	};
+
+	disableRecordAction = {
+		icon: (
+			<MaterialIcon
+				icon="do_not_disturb"
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( 'Disable' ),
+		callback: ( record ) => this.deleteDns( record, 'disable' ),
+	};
+
+	enableRecordAction = {
+		icon: (
+			<Icon
+				icon={ redo }
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( 'Enable' ),
+		callback: ( record ) => this.addDns( record ),
+	};
+
+	recordInfoAction = {
+		icon: (
+			<Icon
+				icon={ info }
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( "What's this?" ),
+		callback: () => {},
+	};
+
+	editRecordAction = {
+		icon: (
+			<Icon
+				icon={ edit }
+				size={ 18 }
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( 'Edit' ),
+		callback: () => {}, // TODO: Add this once the DNS add page is complete
+	};
+
+	deleteRecordAction = {
+		icon: (
+			<Icon
+				icon={ trash }
+				size={ 18 }
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( 'Delete' ),
+		callback: ( record ) => this.deleteDns( record ),
+	};
+
+	deleteDns = ( record, action = 'delete', confirmed = false ) => {
+		const { selectedDomainName, translate } = this.props;
+		const { records } = this.props.dns;
+
+		if ( ! confirmed && isDeletingLastMXRecord( record, records ) ) {
+			this.openDialog( 'deleteEmailForwards', ( result ) => {
+				if ( result.shouldDeleteEmailForwards ) {
+					this.deleteDns( record, action, true );
+				}
+			} );
+
+			return;
+		}
+
+		const successMessage =
+			'delete' === action
+				? translate( 'The DNS record has been deleted.' )
+				: translate( 'The DNS record has been disabled.' );
+
+		const errorMessage =
+			'delete' === action
+				? translate( 'The DNS record has not been deleted.' )
+				: translate( 'The DNS record has not been disabled.' );
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'The DNS record has not been added.' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/dns-records-page-design)